### ISSUE TITLE
Function to turn tables into cross2 object

### ISSUE THREAD

**Issue by kbroman** [rqtl/qtl2#12]
_From @kbroman on November 10, 2016 13:47_

Function to create a cross2 object from data that is already sitting within R. (Currently, the only way to create a cross2 object is with `read_cross2`, with data coming from disk.

_Copied from original issue: rqtl/qtl2geno#103_

**Comment by tavareshugo** [comment]
Dear @kbroman,

I really like this idea, and related to it I think it would be useful to have a helper function to add phenotypes to an existing `cross2` object. [Here's a prototype](https://gist.github.com/tavareshugo/b10c3dca303c28b2d161e439a6ffcac6), in case you'd consider something like it (feel free to use it, change it, or I can fork and pull request if you think this would fit in the package).

My argument for this is that genotypes don't really change across experiments, whereas trait data does. So this would allow one to distribute `cross2` objects (such as in your [qtl2data repo](https://github.com/rqtl/qtl2data) or as data packages) without a `pheno` slot, and the user could later add their own trait data to it. 

cheers,
hugo

**Comment by kbroman** [comment]
Thanks for the suggestion. The `pheno` slot isn't actually required, and the phenotypes are pretty much used separately (e.g., in `scan1()`) so there shouldn't be a need to be able to slot phenotypes in and out.

**Comment by tavareshugo** [comment]
Yes, I see what you mean. I feel like it might still be useful to have some sort of helper function to coerce a `data.frame` to a `matrix` that conforms to the format required by `scan1()` etc. 

Tipically my trait data would start as a `data.frame` (or these days even as a `tibble` or `data.table`), and as I understand it, one has to at least add IDs as rownames and remove the ID column (and any other non-numeric columns). I guess the idea with such a function would be to do that work and check for common genotypes and perhaps warn if not all phenotyped individuals are genotyped. 

Anyway, just a thought, I'm happy to also source my gist whenever I need to use it! :smile:
